File: CPP/Komponen.cpp
#pragma once
#include <iostream>
#include <string>

using namespace std;

// Kelas Komponen sebagai kelas dasar (base class)
class Komponen {
private:
    string merk;
    string nama;

public:
    Komponen()
    {

    }

      // Konstruktor dengan parameter untuk menginisialisasi merk dan nama
    Komponen(string merk, string nama)
    {
        this->merk = merk;
        this->nama = nama;
    }

    //setter
    void setMerk(string merk)
    {
        this->merk = merk;
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    //getter
    string getMerk()
    {
        return merk;
    }

    string getNama()
    {
        return nama;
    }

    ~Komponen()
    {
        
    }
};

File: python/Komponen.py
#deklarasi kelas
class Komponen():
    #atribut private
    __merk = ""
    __nama = ""

    #constuctor
    def __init__(self, merk = "", nama = ""):
        self._merk = merk
        self._nama = nama

    #setter
    def set_merk(self, merk):
        self._merk = merk

    def set_nama(self, nama):
        self._nama = nama

    #getter
    def get_merk(self):
        return self._merk

    def get_nama(self):
        return self._nama